### PR TITLE
Add new `Lint/DuplicateRegexpCharacterClassElement` cop

### PR DESCRIPTION
**Pull request by owst**
Cop documentation:
```
      # This cop checks for duplicate elements in Regexp character classes.
      #
      # @example
      #
      #   # bad
      #   r = /[xyx]/
      #
      #   # bad
      #   r = /[0-9x0-9]/
      #
      #   # good
      #   r = /[xy]/
      #
      #   # good
      #   r = /[0-9x]/
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by marcandre — COMMENTED** [review]
Looks good. Only thing I can see is I think this would trigger a false positive on repeated space:
```ruby
hex = /[
  a-f  # lowercase letter
  0-9 # digit
]+/x
```

File: config/default.yml [hunk 1]
```diff
@@ -2951,6 +2951,11 @@ Style/DoubleNegation:
     - allowed_in_returns
     - forbidden
 
+Style/DuplicateRegexpCharacterClassElement:
+  Description: 'Checks for duplicate elements in Regexp character classes.'
+  Enabled: pending
+  VersionAdded: '0.94'
```
Comment (bbatsov) [inline config/default.yml:2957]: After rebasing update this to `1.1`. :-)

File: config/default.yml [hunk 2]
```diff
@@ -2951,6 +2951,11 @@ Style/DoubleNegation:
     - allowed_in_returns
     - forbidden
 
+Style/DuplicateRegexpCharacterClassElement:
```
Comment (bbatsov) [inline config/default.yml:2954]: Isn't this supposed to be a Lint cop? :-)

Comment (owst) [reply config/default.yml:2954]: 🤦

File: config/default.yml (hunk 1 above)
Comment (owst) [reply config/default.yml:2957]: 🎉 amazing - 1.0! Great news @bbatsov - thank you for all of your and the core team's hard work!

**Comment by owst** [comment]
@marcandre regarding your comment:

> Looks good. Only thing I can see is I think this would trigger a false positive on repeated space:
> 
> ```ruby
> hex = /[
>   a-f  # lowercase letter
>   0-9 # digit
> ]+/x
> ```

Indeed, I thought the same would happen, but actually, it appears that Ruby's free-space mode doesn't support comments inside characters classes (at least on 2.6.6):
```
[1] pry(main)> hex = /[
[1] pry(main)*   a-f  # lowercase letter
[1] pry(main)*   0-9 # digit
[1] pry(main)* ]+/x
=> /[
  a-f  # lowercase letter
  0-9 # digit
]+/x
[2] pry(main)> hex.match('loot')
=> #<MatchData "loot">
```

We shouldn't be able to match `loot` unless `# lowercase letter` is not a comment.

**Review by bbatsov — APPROVED** [review]

**Comment by marcandre** [comment]
Awesome, thanks!

**Pull request merged**